docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,247 @@
+
+The existing `PersistData` setting will be ignored by the code path subscribing to Events since the only reason to do this is to persist the Events. 
+
+There is a race condition for `Marked As Pushed` when Core Data is persisting Events received from the MessageBus. Core Data may not have finished persisting an Event before the Application Service has processed the Event and requested the Event be `Marked As Pushed`. It was decided to remove `Mark as Pushed` capability and just rely on time based scrubbing of old Events.
+
+
+### V2 Event DTO
+
+As this development will be part of the Ireland release all Events published to the MessageBus will use the V2 Event DTO. This is already implemented in Core Data for the V2 AddEvent API.
+
+#### Validation
+
+Services receiving the Event DTO from the MessageBus will log validation errors and stop processing the Event.
+
+### Message Envelope
+
+EdgeX Go Services currently uses a custom Message Envelope for all data that is published to the MessageBus. This envelope wraps the data with metadata, which is `ContentType` (JSON or CBOR), `Correlation-Id` and the obsolete `Checksum`. The `Checksum` is used when the data is CBOR encoded to identify the Event in V1 API to be mark it as pushed. This checksum is no longer needed as the V2 Event DTO requires the ID be set by the Device Services which will always be used in the V2 API to mark the Events as pushed. The Message Envelope will be updated to remove this property.
+
+The C SDK will recreate this Message Envelope.
+
+### Application Services
+
+As part of the V2 API consumption work in Ireland the App Services SDK will be changed to expect to receive V2 Event DTOs rather than the V1 Event model. It will also be updated to no longer expect or use the `Checksum` currently on the  Message Envelope. Note these changes must occur for the V2 consumption and are not directly tied to this effort. 
+
+The App Service SDK will be enhanced for the secure MessageBus connection described below. See **[Secure Connections](#secure-connections)** for details
+
+### MessageBus Topics
+
+> *Note: The change recommended here is not required for this design, but it provides a good opportunity to adopt it.*
+
+Currently Core Data publishes Events to the simple `events` topic. All Application Services running receive every Event published, whether they want them or not. The Events can be filtered out using the `FilterByDeviceName` pipeline function, but the Application Services still receives every Event and process all the Events to some extent. This could cause load issues in a deployment with many devices and large volume of Events from various devices or a very verbose device that that Application Services is not interested in.
```

Comment:
that that Application Services is not --> that **the** Application Services is not